addons/payment/models/payment_acquirer.py

```diff
@@ -251,8 +281,14 @@ def create(self, values_list):
         return acquirers
 
     def write(self, values):
+        enabled_acquirers = self.filtered(lambda acq: acq.state != 'disabled')
         result = super().write(values)
         self._check_required_if_provider()
+
+        # Handle acquirer disabling.
+        if 'state' in values and enabled_acquirers:
```

Comment:
```suggestion
        if 'state' in values and values['state'] == 'disabled' and enabled_acquirers:
```

```acquirer.write({'state': 'enabled'})``` shouldn't disable the tokens if the acquirer is already enabled ;)